src/equations/ideal_glm_mhd_multiion.jl

```diff
@@ -0,0 +1,257 @@
+                      "rho_v3_" * string(i), "rho_e_" * string(i))...)
+    end
+    cons = (cons..., "psi")
+
+    return cons
+end
+
+function varnames(::typeof(cons2prim), equations::AbstractIdealGlmMhdMultiIonEquations)
+    prim = ("B1", "B2", "B3")
+    for i in eachcomponent(equations)
+        prim = (prim...,
+                tuple("rho_" * string(i), "v1_" * string(i), "v2_" * string(i),
+                      "v3_" * string(i), "p_" * string(i))...)
+    end
+    prim = (prim..., "psi")
+
+    return prim
+end
+
+function default_analysis_integrals(::AbstractIdealGlmMhdMultiIonEquations)
+    (entropy_timederivative, Val(:l2_divb), Val(:linf_divb))
+end
+
+"""
+    source_terms_lorentz(u, x, t, equations::AbstractIdealGlmMhdMultiIonEquations)
+
+Source terms due to the Lorentz' force for plasmas with more than one ion species. These source 
+terms are a fundamental, inseparable part of the multi-ion GLM-MHD equations, and vanish for 
+a single-species plasma. In particular, they have to be used for every
+simulation of [`IdealGlmMhdMultiIonEquations2D`](@ref).
+"""
+function source_terms_lorentz(u, x, t, equations::AbstractIdealGlmMhdMultiIonEquations)
+    @unpack charge_to_mass = equations
+    B1, B2, B3 = magnetic_field(u, equations)
+    v1_plus, v2_plus, v3_plus, vk1_plus, vk2_plus, vk3_plus = charge_averaged_velocities(u,
+                                                                                         equations)
+
+    s = zero(MVector{nvariables(equations), eltype(u)})
+
+    for k in eachcomponent(equations)
+        rho, rho_v1, rho_v2, rho_v3, rho_e = get_component(k, u, equations)
+        rho_inv = 1 / rho
+        v1 = rho_v1 * rho_inv
+        v2 = rho_v2 * rho_inv
+        v3 = rho_v3 * rho_inv
+        v1_diff = v1_plus - v1
+        v2_diff = v2_plus - v2
+        v3_diff = v3_plus - v3
+        r_rho = charge_to_mass[k] * rho
+        s2 = r_rho * (v2_diff * B3 - v3_diff * B2)
+        s3 = r_rho * (v3_diff * B1 - v1_diff * B3)
+        s4 = r_rho * (v1_diff * B2 - v2_diff * B1)
+        s5 = v1 * s2 + v2 * s3 + v3 * s4
+
+        set_component!(s, k, 0, s2, s3, s4, s5, equations)
+    end
+
+    return SVector(s)
+end
+
+"""
+    electron_pressure_zero(u, equations::AbstractIdealGlmMhdMultiIonEquations)
+
+Returns the value of zero for the electron pressure. Needed for consistency with the 
+single-fluid MHD equations in the limit of one ion species.
+"""
+function electron_pressure_zero(u, equations::AbstractIdealGlmMhdMultiIonEquations)
+    return zero(u[1])
+end
+
+"""
+    v1, v2, v3, vk1, vk2, vk3 = charge_averaged_velocities(u,
+                                                       equations::AbstractIdealGlmMhdMultiIonEquations)
+
+
+Compute the charge-averaged velocities (`v1`, `v2`, and `v3`) and each ion species' contribution
+to the charge-averaged velocities (`vk1`, `vk2`, and `vk3`). The output variables `vk1`, `vk2`, and `vk3`
+are `SVectors` of size `ncomponents(equations)`.
```

Comment:
Is this comment reflecting what is actually returned? There is more returned than just vectors, right?

Reply:
Yes, it is. Three scalar variables (v1, v2 and v3) and three vectors (vk1, vk2, and vk3)

Reply:
I see - I suggest a clarification such that one does not think only the the vectors are returned.

Reply:
Is it really unclear from the docstring?? One can clearly see that v1, v2 and v3 are returned!

https://github.com/trixi-framework/Trixi.jl/blob/79b0e29dbda74c9ffbbabbaed466c2fdd91414b4/src/equations/ideal_glm_mhd_multiion.jl#L87-L97